### PR TITLE
Refactor config

### PR DESCRIPTION
**Pull request by kiootic**
ref #4

### CODE REVIEW

File: pkg/validation/formats.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	}
+
+	if !f.AllowName && addr.Name != "" {
+		return errors.New("input email must not have name")
+	}
+
+	return nil
+}
+
+// FormatURI checks if input is an absolute URI, without query/fragment.
```
Comment (louischan-oursky) [inline pkg/validation/formats.go:59]: Why `uri` cannot have query or fragment?

File: pkg/auth/config/secret_data.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+`)
+
+type DatabaseCredentials struct {
+	DatabaseURL    string `json:"database_url,omitempty"`
+	DatabaseSchema string `json:"database_schema,omitempty"`
+}
+
+func (c *DatabaseCredentials) SetDefaults() {
+	if c.DatabaseSchema == "" {
+		c.DatabaseSchema = "public"
```
Comment (louischan-oursky) [inline pkg/auth/config/secret_data.go:25]: Let's use `app`?

File: pkg/auth/config/secret_data.go [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+	Host     string   `json:"host,omitempty"`
+	Port     int      `json:"port,omitempty"`
+	Mode     SMTPMode `json:"mode,omitempty"`
+	Username string   `json:"username,omitempty"`
+	Password string   `json:"password,omitempty"`
+}
+
+func (c *SMTPServerCredentials) SetDefaults() {
+	if c.Mode == "" {
+		c.Mode = SMTPModeNormal
```
Comment (louischan-oursky) [inline pkg/auth/config/secret_data.go:123]: Default should be ssl

Comment (kiootic) [reply pkg/auth/config/secret_data.go:123]: It's the same default value as current configuration, do you want to change it?

File: pkg/auth/config/secret_data.go (hunk 2 above)
Comment (kiootic) [reply pkg/auth/config/secret_data.go:25]: `public` schema is the default schema of Postgres, so developer does not need to create a new schema for quick start.

File: pkg/validation/formats.go (hunk 1 above)
Comment (kiootic) [reply pkg/validation/formats.go:59]: This is `URLFullOnly` from the original format. I suppose I'll make it accepts any URI (except URI references).

**Pull request merged**